sycl/include/CL/sycl/handler2.hpp

```diff
@@ -206,94 +211,132 @@ class handler {
+      const int &Size = KernelArgs[I].info;
+      processArg(Ptr, Kind, Size, I, IndexShift, IsKernelCreatedFromSource);
+    }
+  }
 
-      switch (Kind) {
-      case detail::kernel_param_kind_t::kind_std_layout: {
-        const size_t Size = KernelArgs[I].info;
-        MArgs.emplace_back(detail::ArgDesc(Kind, Ptr, Size, NextArgId));
-        break;
-      }
-      case detail::kernel_param_kind_t::kind_accessor: {
-
-        const int AccTarget = KernelArgs[I].info & 0x7ff;
-        switch (static_cast<access::target>(AccTarget)) {
-        case access::target::global_buffer:
-        case access::target::constant_buffer: {
-
-          detail::AccessorBaseHost *AccBase = (detail::AccessorBaseHost *)Ptr;
-
-          detail::Requirement *AccImpl = detail::getSyclObjImpl(*AccBase).get();
-
-          MArgs.emplace_back(
-              detail::ArgDesc(Kind, AccImpl, /*Size=*/0, NextArgId));
-          MRequirements.emplace_back(AccImpl);
-
-          MArgs.emplace_back(
-              detail::ArgDesc(detail::kernel_param_kind_t::kind_std_layout,
-                              &(AccImpl->MAccessRange[0]),
-                              sizeof(size_t) * AccImpl->MDims, NextArgId + 1));
-          MArgs.emplace_back(
-              detail::ArgDesc(detail::kernel_param_kind_t::kind_std_layout,
-                              &AccImpl->MMemoryRange[0],
-                              sizeof(size_t) * AccImpl->MDims, NextArgId + 2));
-          MArgs.emplace_back(
-              detail::ArgDesc(detail::kernel_param_kind_t::kind_std_layout,
-                              &AccImpl->MOffset[0],
-                              sizeof(size_t) * AccImpl->MDims, NextArgId + 3));
-          NextArgId += 3;
-          break;
-        }
-        case access::target::local: {
-
-          detail::LocalAccessorBaseHost *LAcc =
-              (detail::LocalAccessorBaseHost *)Ptr;
-          range<3> &Size = LAcc->getSize();
-          const int Dims = LAcc->getNumOfDims();
-          int SizeInBytes = LAcc->getElementSize();
-          for (int I = 0; I < Dims; ++I)
-            SizeInBytes *= Size[I];
-          MArgs.emplace_back(
-              detail::ArgDesc(detail::kernel_param_kind_t::kind_std_layout,
-                              nullptr, SizeInBytes, NextArgId));
-
-          MArgs.emplace_back(
-              detail::ArgDesc(detail::kernel_param_kind_t::kind_std_layout,
-                              &Size, Dims * sizeof(Size[0]), NextArgId + 1));
-          MArgs.emplace_back(
-              detail::ArgDesc(detail::kernel_param_kind_t::kind_std_layout,
-                              &Size, Dims * sizeof(Size[0]), NextArgId + 2));
-          MArgs.emplace_back(
-              detail::ArgDesc(detail::kernel_param_kind_t::kind_std_layout,
-                              &Size, Dims * sizeof(Size[0]), NextArgId + 3));
-          NextArgId += 3;
-          break;
-        }
+  void extractArgsAndReqs() {
```

Comment:
All this code from here would require some comments so it can be understood...

Reply:
I added a small comment.